src/ring/middleware/session_timeout.clj

```diff
@@ -4,6 +4,40 @@
 (defn- current-time []
   (quot (System/currentTimeMillis) 1000))
 
+(defn- session-expired?
```

Comment:
Let's split this into a an `session-absolute-expired?` and a `session-idle-expired?`.